DOC-8536 initial update of sample-application intro

osfameron opened the pull request:
Each travel-sample app backend app already has good documentation
in its README.md on git, but we need an entrypoint in the SDK docs
too.

This commit:

  * uses Asciidoc attributes to provide a shared skeleton that could be used
  with partials across all the backend apps.
  * mostly defers to the README.
  * has a number of marked TODOs as next steps

*NOTE*: this is a draft, just first commit, to discuss with Maria and Richard!

osfameron committed:
DOC-8536 initial update of sample-application intro

Each travel-sample app backend app already has good documentation
in its README.md on git, but we need an entrypoint in the SDK docs
too.

This commit:

  * uses Asciidoc attributes to provide a shared skeleton that could be used
  with partials across all the backend apps.
  * mostly defers to the README.
  * has a number of marked TODOs as next steps

maria-robobug commented:
Nice! 

I like how you've parameterized things so that this document can be more generic, could this live in `sdk-common` somehow?

If it's doable then we would just have to reference `sdk-common` when updating all the other SDK sample app pages 🤔

RichardSmedley commented on modules/hello-world/pages/sample-application.adoc:
```diff
@@ -5,69 +5,77 @@
-set up as described above, you will need `git` to fetch the travel sample application code:
-
-[source,bash]
+[source,bash,subs="+attributes"]
```
Needs to be console,
like in the other docs - 
bash highlighting is geared towards .sh scripts  ;)

RichardSmedley commented on modules/hello-world/pages/sample-application.adoc:
```diff
@@ -5,69 +5,77 @@
+:travel-sample-git-project: try-cb-nodejs
+:travel-sample-entrypoint: index.js
+
+// TODO: move these to attributes.adoc
```
:+1:

osfameron commented:
> I like how you've parameterized things so that this document can be more generic, could this live in `sdk-common` somehow?
Yes! that's the plan, but seemed clearer to keep in one repo for discussion for first cut, to see if we want to do it.

RichardSmedley reviewed (commented):
LGTM
Will need a script to regularly copy to README (or a Git Action...)
as the other repo doesn't speak Antora.
:+1:

RichardSmedley commented:
> Will need a script to regularly copy to README (or a Git Action...)
> as the other repo doesn't speak Antora.

That is, if we're going to do that as well :)

osfameron commented:
> Will need a script to regularly copy to README (or a Git Action...)
> as the other repo doesn't speak Antora.

Ooo, interesting - I think the docs entrypoint is slightly different from the github project README? But that's an option I guess, if we fleshed this out to include the same level of detail.

RichardSmedley commented:
> Ooo, interesting - I think the docs entrypoint is slightly different from the github project README?
> But that's an option I guess, if we fleshed this out to include the same level of detail.

Just a passing thought, as everything's getting nicely reusable,
but no need to fix the whole universe at once 
*Maybe :D

maria-robobug commented:
Yeah I would also agree that the Github project README would be slightly different from the doc itself. But we can review that in future for sure.

osfameron committed:
Update attributes (all point to project-docs)

osfameron committed:
Extract sample-applications partials to sdk-common

Requires https://github.com/couchbase/docs-sdk-common/pull/23

Note that the Sample App Backend section is not extracted
for now, as this requires further work, and is also highly
specific to each app version.

osfameron commented:
Requires https://github.com/couchbase/docs-sdk-common/pull/23

osfameron committed:
Merge branch 'release/3.2' into DOC-8536-sample-application-docs
